modules/spring-data-2.2-ext/src/main/java/org/apache/ignite/springdata22/repository/support/IgniteRepositoryFactory.java

```diff
@@ -219,12 +235,19 @@ private IgniteCache getRepositoryCache(Class<?> repoIf) {
                 boolean annotatedIgniteQuery = !annotation.dynamicQuery() && (StringUtils.hasText(qryStr) || annotation
                     .textQuery());
 
+                boolean isTextQry = annotation.textQuery();
+
+                if (isTextQry && isThinClient) {
+                    throw new IllegalStateException(
```

Comment:
Do we really need to fail here? Perhaps we can do this check in runtime in IgniteRepositoryQuery#execute, in this case we can use common repository for thin and thick client. WDYT?

Reply:
I think, you are right and it's enough to mention this behavior in the documentation / javadocs.

Reply:
Done.